### PR TITLE
Create GP from prior with mean factory

### PR DESCRIPTION
**Pull request by kalama-ai**
New constructor method that enables transfer learning for Gaussian Process surrogates

```
# Train source GP
source_gp = GaussianProcessSurrogate()
source_gp.fit(source_space, source_objective, source_data)

# Transfer mean
target_gp = GaussianProcessSurrogate.from_prior(
      prior_gp=source_gp,  # Use source GP as prior
  )
```

Mean Function Transfer:
  - Implements full mean transfer from the prior GP to the new GP
  - The posterior mean predictions of the pre-trained GP used as the mean module for the new GP
  - Frozen hyperparameters 
  - Mean function is evaluated at source training points
  - New `PriorMeanFactory` in the GP component factory system (lazily constructs a PriorMean module)

### CODE REVIEW

File: baybe/surrogates/gaussian_process/components/mean.py [hunk 1]
```diff
@@ -36,3 +39,42 @@ def __call__(
...
+                self.gp: SingleTaskGP = deepcopy(gp)
+                for param in self.gp.parameters():
+                    param.requires_grad = False
+
+            def forward(self, x: Tensor) -> Tensor:
+                """Compute the mean using the wrapped GP's posterior."""
+                self.gp.eval()
+                self.gp.likelihood.eval()
+                with torch.no_grad(), gpytorch.settings.fast_pred_var():
+                    return self.gp(x).mean.detach()
```
[the remark this answers is not in the document]
Comment (kalama-ai) [reply baybe/surrogates/gaussian_process/components/mean.py]: Actually, double normalization is not the issue here. The issue here is that the mean of the prior GP is evaluated on the normalized input (normalized by the new GP's data). The prior mean should be evaluated at the unnormalized data. To this end we need to recompute the normalization in the new factory and unnormalize the data before evaluating. This has been fixed in this PR: 177ad21827982397ff254809f6c3ebf09ac17b85. Tests have been added.

[the remark this answers is not in the document]
Comment (kalama-ai) [reply baybe/surrogates/gaussian_process/components/mean.py]: Removed `no_grad` and `detach` so input gradients flow through during `optimize_acqf`. The prior GP's params are still frozen. This only mattered for continuous/mixed spaces. Discrete was unaffected since it just evaluates candidates directly.

File: baybe/surrogates/gaussian_process/components/mean.py [hunk 2]
```diff
@@ -36,3 +39,42 @@ def __call__(
...
+
+
+@define
+class PriorMeanFactory(MeanFactoryProtocol):
+    """A factory that creates a prior mean from a trained botorch GP model.
+
+    The mean function uses the trained GP's posterior mean predictions.
+    The provided model is deep-copied and its parameters are frozen.
+    """
+
```
[the remark this answers is not in the document]
Comment (kalama-ai) [reply baybe/surrogates/gaussian_process/components/mean.py:59]: Added hook.

**Comment by kalama-ai** [comment]
Will be converted in a posterior mean method.
